src/Messenger/Clients/Sns.php

```diff
@@ -176,10 +178,10 @@ public function confirmSubscription(string $topic, string $token): bool
         return true;
     }
 
-    public function unsubscribe(string $topic, string $subscriptionId): bool
+    public function unsubscribe(string $subscriptionArn): bool
     {
         $this->snsClient->unsubscribe([
-            'SubscriptionArn' => $this->getArnFromName($topic).':'.$subscriptionId,
+            'SubscriptionArn' => $subscriptionArn,
```

Comment:
@claudioemmanuel adiciona o $subscriptionArn como parâmetro e mantem os 2 anteriores pra não quebrar o contrato existente pra outras aplicações.

`public function unsubscribe(string $topic = null, string $subscriptionId = null, string $subscriptionArn = null):
$arn = subscriptionArn ? subscriptionArn : $this->getArnFromName($topic).':'.$subscriptionId;
`